lib/integrations/externalcloudaudit/bootstrap.go

```diff
@@ -0,0 +1,405 @@
+	err := createBucket(ctx, clt, bucketName, region, false)
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err)
+		}
+	}
+
+	_, err = clt.PutBucketLifecycleConfiguration(ctx, &s3.PutBucketLifecycleConfigurationInput{
+		Bucket: &bucketName,
+		LifecycleConfiguration: &s3types.BucketLifecycleConfiguration{
+			Rules: []s3types.LifecycleRule{
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireLargePayloads"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 4,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/large_payloads",
+					},
+				},
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireQueryResults"),
+					Expiration: &s3types.LifecycleExpiration{
+						Days: 1,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{
+						Value: "/query_results",
+					},
+				},
+				{
+					Status: s3types.ExpirationStatusEnabled,
+					ID:     aws.String("ExpireNonCurrentVersionsAndDeleteMarkers"),
+					NoncurrentVersionExpiration: &s3types.NoncurrentVersionExpiration{
+						NewerNoncurrentVersions: 0,
+						NoncurrentDays:          1,
+					},
+					AbortIncompleteMultipartUpload: &s3types.AbortIncompleteMultipartUpload{
+						DaysAfterInitiation: 7,
+					},
+					Expiration: &s3types.LifecycleExpiration{
+						ExpiredObjectDeleteMarker: true,
+					},
+					Filter: &s3types.LifecycleRuleFilterMemberPrefix{},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
+}
+
+func createBucket(ctx context.Context, clt createBucketClient, bucketName string, region string, objectLock bool) error {
+	_, err := clt.CreateBucket(ctx, &s3.CreateBucketInput{
+		Bucket: &bucketName,
+		CreateBucketConfiguration: &s3types.CreateBucketConfiguration{
+			LocationConstraint: s3types.BucketLocationConstraint(region),
+		},
+		ObjectLockEnabledForBucket: objectLock,
+		ACL:                        s3types.BucketCannedACLPrivate,
+		ObjectOwnership:            s3types.ObjectOwnershipBucketOwnerEnforced,
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	_, err = clt.PutBucketVersioning(ctx, &s3.PutBucketVersioningInput{
+		Bucket: &bucketName,
+		VersioningConfiguration: &s3types.VersioningConfiguration{
+			Status: s3types.BucketVersioningStatusEnabled,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(awsutil.ConvertS3Error(err))
+	}
+
+	return nil
+}
+
+// createAthenaWorkgroupClient describes the required methods to ensure and configure athena workgroups for external cloud audit
+type createAthenaWorkgroupClient interface {
+	// Creates a workgroup with the specified name.
+	CreateWorkGroup(ctx context.Context, params *athena.CreateWorkGroupInput, optFns ...func(*athena.Options)) (*athena.CreateWorkGroupOutput, error)
+}
+
+// ensureAthenaWorkgroup ensures an athena workgroup in which to run the athena sql queries in
```

Comment:
```suggestion
// ensureAthenaWorkgroup ensures an athena workgroup in which to run the athena sql queries
```

Reply:
Updated in [eee2744](https://github.com/gravitational/teleport/pull/33272/commits/eee274459cb8ab12b7ddc1b6701ae5690d03f3f4)